### PR TITLE
Add ability to override module's info #1421

### PR DESCRIPTION
**Pull request by iliarostovtsev**
https://github.com/virtualmin/virtualmin-gpl/issues/275
https://github.com/webmin/webmin/issues/1421#issuecomment-772137324

### CODE REVIEW

File: web-lib-funcs.pl [hunk 1]
```diff
@@ -5639,19 +5639,19 @@ sub get_module_overrides
...
+if (-r "$mdir/module.overrides") {
 	my %overrides;
 	&read_file_cached("$mdir/module.overrides", \%overrides);
 	my $funcs = $overrides{'funcs'};
 	if ($funcs) {
 		eval {
 			local $main::error_must_die = 1;
 			my @funcs = split(/\s+/, $funcs);
-			&foreign_require($mod);
+			&foreign_require($mod, "$mod-lib.pl");
```
Comment (iliarostovtsev) [inline web-lib-funcs.pl:5650]: This is an important fix as we need to define module's lib manually as `foreign_require` will call `get_module_info` in auto detection mode causing endless loop.

We should assume any module's overrides sub reside in main module's lib file.

**Comment by jcameron** [comment]
I like this solution, but it could be simplified by instead of having the override function be defined in `module.overrides` and implemented in the main -lib.pl file, do it in a separate file. Sort of like how modules have a `backup_config.pl` file already.

**Comment by iliarostovtsev** [comment]
Okay, simplification is done. https://github.com/webmin/webmin/pull/1470/files

File: web-lib-funcs.pl [hunk 2]
```diff
@@ -5622,6 +5625,30 @@ sub get_module_info
...
+
+my $mdir = &module_root_directory($mod);
+
+# Call module specific overrides
+my $call = 'module_overrides';
+if (-r "$mdir/$call.pl") {
+	eval {
+		local $main::error_must_die = 1;
+		&foreign_require($mod, "$call.pl");
+		&foreign_call($mod, $call, \%{$data});
```
Comment (jcameron) [inline web-lib-funcs.pl:5647]: Why \%{$data} rather than just $data ?

Comment (iliarostovtsev) [reply web-lib-funcs.pl:5647]: It wouldn't alter the data, if not passed as reference?

Comment (jcameron) [reply web-lib-funcs.pl:5647]: Yes, but isn't the whole idea to alter the data?

Comment (iliarostovtsev) [reply web-lib-funcs.pl:5647]: 🙂 Yes, the whole point is to alter the data and passing it as `$data` the second time wouldn't work. No?

Comment (jcameron) [reply web-lib-funcs.pl:5647]: Since $data is a hash reference, you can alter the contents by just passing it as $data

Comment (iliarostovtsev) [reply web-lib-funcs.pl]: Actually you're right! Fixed! Thanks for the review.

**Comment by jcameron** [comment]
Looks good, thanks!

**Comment by iliarostovtsev** [comment]
Cool, thanks.

**Pull request merged**